File: Projects/Framework-Samples/Csla/CSharp/PetShop.Silverlight/Source/PetShop.UI.Silverlight/Models/CategoryDetailModel.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

using PetShop.Business;

namespace PetShop.UI.Silverlight
{
    public class CategoryDetailModel : FrameworkElement
    {
        public static readonly DependencyProperty SelectedItemsProperty =
            DependencyProperty.Register("SelectedItems", typeof(List<Category>), typeof(CategoryDetailModel), null);

        public List<Category> SelectedItems
        {
            get { return (List<Category>)GetValue(SelectedItemsProperty); }
            set { SetValue(SelectedItemsProperty, value); }
        }

        public void Home()
        {
            MainPageModel.ShowForm(new CategoryListPage());
        }
    }
}

File: tags/CSLA-4.0-Samples/VisualBasic/PetShop/Source/PetShop.UI.Silverlight/Models/CategoryDetailModel.vb
Imports System.Collections.Generic
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes

Imports PetShop.Business

Public Class CategoryDetailModel
    Inherits FrameworkElement
    Public Shared ReadOnly SelectedItemsProperty As DependencyProperty = DependencyProperty.Register("SelectedItems", GetType(List(Of Category)), GetType(CategoryDetailModel), Nothing)
    Public Property SelectedItems() As List(Of Category)
        Get
            Return DirectCast(GetValue(SelectedItemsProperty), List(Of Category))
        End Get
        Set(ByVal value As List(Of Category))
            SetValue(SelectedItemsProperty, value)
        End Set
    End Property

    Public Sub Home()
        MainPageModel.ShowForm(New CategoryListPage())
    End Sub
End Class